### PR TITLE
[267] Display invalid representations when viewpoint is not loaded

### PR DESCRIPTION
**Pull request by lfasani**
It also manages the case when the representation description is not loaded

Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/267

### CODE REVIEW

File: plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/ui/tools/internal/views/common/item/RepresentationDescriptionItemImpl.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public Object getWrappedObject() {
 
     @Override
     public int compareTo(final RepresentationDescriptionItemImpl o) {
-        if (representationDescription.getName() != null) {
+        if (representationDescription != null && representationDescription.getName() != null) {
             return representationDescription.getName().compareTo(o.representationDescription.getName());
```
Comment (scosta-obeo) [inline plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/ui/tools/internal/views/common/item/RepresentationDescriptionItemImpl.java:127]: o.representationDescription can also be null

File: plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/ui/tools/internal/views/common/item/ViewpointItemImpl.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ public Object getWrappedObject() {
 
     @Override
     public int compareTo(final ViewpointItemImpl o) {
-        if (viewpoint.getName() != null) {
+        if (viewpoint != null && viewpoint.getName() != null) {
             return viewpoint.getName().compareTo(o.viewpoint.getName());
```
Comment (scosta-obeo) [inline plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/ui/tools/internal/views/common/item/ViewpointItemImpl.java:98]: o.viewpoint can be null

File: plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/viewpoint/provider/Messages.java
Comment (scosta-obeo) [inline plugins/org.eclipse.sirius.ui/src/org/eclipse/sirius/viewpoint/provider/Messages.java:1]: Copyright 2024

**Review by scosta-obeo — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.ui/icons/full/obj16/diagram-type.gif
[the remark this answers is not in the document]
Comment (lfasani) [reply plugins/org.eclipse.sirius.ui/icons/full/obj16/diagram-type.gif:1]: yes but not in this commit

**Pull request merged**